### DISCUSSION TITLE
Query metadata through an external API in the compiled site

### DISCUSSION THREAD

**Discussion by yl5682** [thegetty/quire#781] [Development]
I was prompted to have this question when I was suggested by Harvard Art Museums to query their APIs to get the latest metadata to art objects that they curate. The idea is that the metadata may change from time to time as they update the information. Instead of baking the metadata into a Quire project's `objects.yml`, I am encouraged to make a call to an API like this:
`https://api.harvardartmuseums.org/object/123456`
and extract the metadata from corresponding fields from the JSON response. 

However, this seems to conflict the idea that a Quire site is static, and is generated at the build time. 

Therefore I want to get some confirmation (or opposition) from the maker of Quire - should I gather metadata and have them integrated into the corresponding ymls during build time? Or is there an optimal way to let a built Quire site still pull data through an API live when a user is accessing the site?

Please feel free to advise/discuss. I look forward to best practice advice!

Thank you!

**Comment by Erin-Cecele** [comment]
Hi @yl5682 thank you for this question. We will be in touch with feedback shortly!

**Reply by Erin-Cecele** [reply]
Apologies for the delay @yl5682. This is absolutely still on our radar. We are working on some tight deadlines but will come back to this inquiry. Thank you for your patience.

**Reply by yl5682** [reply]
Thank you for the update and I appreciate your assistance!
